### ISSUE TITLE
Marking a cop as unsafe should imply that its auto-correct is unsafe as well

### ISSUE THREAD

**Issue by buehmann** [rubocop-hq/rubocop#7611]
We found this issue when discussing https://github.com/rubocop-hq/rubocop/pull/7307.

There are two annotations that mark aspects of a cop as (un)safe:
> - Safe (true/false) - indicates whether the cop can yield false positives (by
>   design) or not.
> - SafeAutoCorrect (true/false) - indicates whether the auto-correct the cop
>   does is safe (equivalent) by design.

https://github.com/rubocop-hq/rubocop/blob/23d5e4b7eca052eaf25c63b818ed338f976162bb/manual/auto_correct.md#safe-auto-correct

There are command-line options that select cops and auto-corrections based on those flags:
```
        --safe                       Run only safe cops.
        --safe-auto-correct          Run auto-correct only when it's safe.
```

## Expected behavior

Our assumption is that a cop that is marked as not `Safe` cannot have a safe auto-correction. Otherwise auto-correction would be applied to false positives as well and the result would be unpredictable.

That means that `--safe-auto-correct` must not run auto-correction for cops with `Safe: false`.

## Actual behavior

`--safe-auto-correct` runs auto-correction for unsafe cops.

## Steps to reproduce the problem

Pick a cop that only has `Safe: false`, but not `SafeAutoCorrect: false` in the default config:
https://github.com/rubocop-hq/rubocop/blob/8d071f0c9a32f594b696cb7a3b6a0e5adabad42b/config/default.yml#L3344-L3354
```sh-session
$ cat example.rb          
# frozen_string_literal: true

# thief is not a Hash
thief.has_key?(:foo)

$ rubocop -f c example.rb 
example.rb:4:7: C: Style/PreferredHashMethods: Use Hash#key? instead of Hash#has_key?.
thief.has_key?(:foo)
      ^^^^^^^^

1 file inspected, 1 offense detected

$ rubocop -f c --safe example.rb

1 file inspected, no offenses detected

$ rubocop -f c --safe-auto-correct example.rb
example.rb:4:7: C: [Corrected] Style/PreferredHashMethods: Use Hash#key? instead of Hash#has_key?.
thief.has_key?(:foo)
      ^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```
Here the (wrong) autocorrect is applied although the cop is supposed to be safe.
